Mathlib/Combinatorics/SimpleGraph/CompleteMultipartite.lean

```diff
@@ -162,4 +169,121 @@ theorem IsCompleteMultipartite.comap {β : Type*} {H : SimpleGraph β} (f : H 
   exact not_isCompleteMultipartite_of_pathGraph3ComplEmbedding
           <| f.comp (pathGraph3ComplEmbeddingOf h)
 
+section CompleteEquipartiteGraph
+
+variable {r t : ℕ}
+
+/-- The **complete equipartite graph** in `r` parts each of *equal* size `t` such that two
+vertices are adjacent if and only if they are in different parts. -/
+abbrev completeEquipartiteGraph (r t : ℕ) : SimpleGraph ((Fin r) × (Fin t)) :=
+  (⊤ : SimpleGraph (Fin r)).comap Prod.fst
+
+/-- A `completeEquipartiteGraph` is isomorphic to a corresponding `completeMultipartiteGraph`.
+
+The difference is that the former vertices are a product type whereas the latter vertices are a
+dependent product type. -/
+def completeEquipartiteGraph.completeMultipartiteGraph :
+    completeEquipartiteGraph r t ≃g completeMultipartiteGraph (Function.const (Fin r) (Fin t)) where
+  toFun := fun (v₁, v₂) ↦ by
+    use v₁, v₂, v₂.is_lt
+  invFun := fun ⟨v₁, v₂⟩ ↦ by
+    rw [Function.const_apply] at v₂
```

Comment:
It's usually not a good idea to use tactics to create a function like this; I wonder if you can just remove this line?